File: codecamp/src/main/java/com/gdogaru/codecamp/ApplicationModule.java
package com.gdogaru.codecamp;

import android.app.Application;
import android.content.Context;

import dagger.Module;
import dagger.Provides;

/**
 *
 */
@Module
public class ApplicationModule {
    App application;

    public ApplicationModule(App application) {
        this.application = application;
    }

    @Provides
    App app() {
        return application;
    }

    @Provides
    Application application() {
        return application;
    }

    @Provides
    Context context() {
        return application;
    }
}

File: codecamp/src/main/java/com/gdogaru/codecamp/di/ApplicationModule.kt
/*
 * Copyright (c) 2019 Gabriel Dogaru - gdogaru@gmail.com
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *  *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package com.gdogaru.codecamp.di

import android.app.Application
import android.content.Context
import com.gdogaru.codecamp.App
import dagger.Module
import dagger.Provides
import javax.inject.Inject
import javax.inject.Singleton

/**
 *
 */
@Module
class ApplicationModule
@Inject constructor(var application: App) {

    @Singleton
    @Provides
    internal fun application(): Application = application

    @Singleton
    @Provides
    internal fun context(): Context = application

    @Singleton
    @Provides
    internal fun app(): App = application

}
